test/e2e/preemption/preemption_test.go

```diff
@@ -0,0 +1,337 @@
+ limitations under the License.
+*/
+
+package preemption_test
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/apache/yunikorn-k8shim/pkg/common/constants"
+	tests "github.com/apache/yunikorn-k8shim/test/e2e"
+	"github.com/apache/yunikorn-k8shim/test/e2e/framework/configmanager"
+	"github.com/apache/yunikorn-k8shim/test/e2e/framework/helpers/common"
+	"github.com/apache/yunikorn-k8shim/test/e2e/framework/helpers/k8s"
+	"github.com/apache/yunikorn-k8shim/test/e2e/framework/helpers/yunikorn"
+)
+
+var kClient k8s.KubeCtl
+var restClient yunikorn.RClient
+var ns *v1.Namespace
+var dev = "dev" + common.RandSeq(5)
+var oldConfigMap = new(v1.ConfigMap)
+var annotation = "ann-" + common.RandSeq(10)
+
+// Nodes
+var Worker = ""
+var WorkerMemRes int64
+var sleepPodMemLimit int64
+var taintKey = "e2e_test_preemption"
+var nodesToTaint []string
+
+// Queues
+var preemptionQueueYamlFormat = `
+partitions:
+  - name: default
+    queues:
+      - name: root
+        submitacl: '*'
+        queues:
+          - name: sandbox1
+            submitacl: '*'
+            properties:
+              preemption.delay: 1s
+            resources:
+              guaranteed:
+                memory: %dM
+          - name: sandbox2
+            submitacl: '*'
+            properties:
+              preemption.policy: %s
+              preemption.delay: 1s
+            resources:
+              guaranteed:
+                memory: %dM
+`
+
+var _ = ginkgo.BeforeSuite(func() {
+	// Initializing kubectl client
+	kClient = k8s.KubeCtl{}
+	Ω(kClient.SetClient()).To(gomega.BeNil())
+	// Initializing rest client
+	restClient = yunikorn.RClient{}
+	Ω(restClient).NotTo(gomega.BeNil())
+
+	annotation = "ann-" + common.RandSeq(10)
+	yunikorn.EnsureYuniKornConfigsPresent()
+	yunikorn.UpdateConfigMapWrapper(oldConfigMap, "", annotation)
+
+	ginkgo.By("Port-forward the scheduler pod")
+	var err = kClient.PortForwardYkSchedulerPod()
+	Ω(err).NotTo(gomega.HaveOccurred())
+
+	ginkgo.By("create development namespace")
+	ns, err = kClient.CreateNamespace(dev, nil)
+	gomega.Ω(err).NotTo(gomega.HaveOccurred())
+	gomega.Ω(ns.Status.Phase).To(gomega.Equal(v1.NamespaceActive))
+
+	var nodes *v1.NodeList
+	nodes, err = kClient.GetNodes()
+	Ω(err).NotTo(gomega.HaveOccurred())
+	Ω(len(nodes.Items)).NotTo(gomega.BeZero(), "Nodes cant be empty")
+
+	// Extract node allocatable resources
+	for _, node := range nodes.Items {
+		// skip master if it's marked as such
+		node := node
+		if k8s.IsMasterNode(&node) || !k8s.IsComputeNode(&node) {
+			continue
+		}
+		if Worker == "" {
+			Worker = node.Name
+		} else {
+			nodesToTaint = append(nodesToTaint, node.Name)
+		}
+	}
+	Ω(Worker).NotTo(gomega.BeEmpty(), "Worker node not found")
+
+	ginkgo.By("Tainting some nodes..")
```

Comment:
If you/we decide to keep this, it makes sense to extract code from this file, because taining/untaining seems to be a copy-paste.

Reply:
I prefer to taint nodes because [cross-node preemption](https://yunikorn.apache.org/docs/next/design/preemption#non-goals) is in non-goals. I added `TaintNodes` and `UntaintNodes` to reduce some duplicated code.